# 0 represents unoccupied
# -1 represents unknown
# > 0 represents occupied
int8[] data